lib/state/appActions.js

```diff
@@ -53,18 +54,50 @@ appActions.errorText = {
 };
 
 appActions.errorStage = {
-  STAGE_SETUP : { Code: 'E_SETUP' , Message: 'Error during setup', Step: '' },
-  STAGE_LOGIN : { Code: 'E_LOGIN' , Message: 'Error during login', Step: '' },
-  STAGE_LOGOUT : { Code: 'E_LOGOUT' , Message: 'Error during logout', Step: '' },
-  STAGE_DETECT_DEVICES : { Code: 'E_DETECTING_DEVICES' , Message: 'Error detecting devices', Step: '' },
-  STAGE_READ_FILE : { Code: 'E_READING_FILE' , Message: 'Error trying to read file' , Step: '' },
-  STAGE_UPLOAD : { Code: 'E_UPLOADING' , Message: 'Error uploading data' , Step: ''},
-  STAGE_DEVICE_UPLOAD : { Code: 'E_DEVICE_UPLOAD' , Message: 'Error uploading device data', Step: '' },
-  STAGE_DEVICE_DETECT : { Code: 'E_DEVICE_DETECT' , Message: 'Error trying to detect a device' , Step: '' },
-  STAGE_CARELINK_UPLOAD : { Code: 'E_CARELINK_UPLOAD' , Message: 'Error uploading CareLink data' , Step: ''},
-  STAGE_CARELINK_FETCH : { Code: 'E_CARELINK_FETCH' , Message: 'Error fetching CareLink data' , Step: '' }
-};
+  STAGE_SETUP : { code: 'E_SETUP' , friendlyMessage: 'Error during setup' },
+  STAGE_LOGIN : { code: 'E_LOGIN' , friendlyMessage: 'Error during login' },
+  STAGE_LOGOUT : { code: 'E_LOGOUT' , friendlyMessage: 'Error during logout' },
+  STAGE_DETECT_DEVICES : { code: 'E_DETECTING_DEVICES' , friendlyMessage: 'Error detecting devices' },
+  STAGE_PICK_FILE : { code: 'E_SELECTING_FILE' , friendlyMessage: 'Error during file selection' },
+  STAGE_READ_FILE : { code: 'E_READING_FILE' , friendlyMessage: 'Error trying to read file' },
+  STAGE_UPLOAD : { code: 'E_UPLOADING' , friendlyMessage: 'Error uploading data' },
+  STAGE_DEVICE_UPLOAD : { code: 'E_DEVICE_UPLOAD' , friendlyMessage: 'Error uploading device data' },
+  STAGE_DEVICE_DETECT : { code: 'E_DEVICE_DETECT' , friendlyMessage: 'Error trying to detect a device' },
+  STAGE_CARELINK_UPLOAD : { code: 'E_CARELINK_UPLOAD' , friendlyMessage: 'Error uploading CareLink data' },
+  STAGE_CARELINK_FETCH : { code: 'E_CARELINK_FETCH' , friendlyMessage: 'Error fetching CareLink data' }
+};
+
+function extractMessage(err) {
+  if (typeof err.message === 'object' && err.message.message) {
+    return err.message.message;
+  }
+  else {
+    return err.message;
+  }
+}
 
+function getErrorName(err) {
+  if (err.name) {
+    return err.name;
+  }
+  return 'POST error';
+}
+
+appActions.addMoreInfoToError = function(err, stage) {
```

Comment:
as below I think we should just pass in the actual stage 

``appActions.addMoreInfoToError(err, appActions.errorStage.STAGE_SETUP);``